xarray/plot/facetgrid.py

```diff
@@ -266,6 +265,46 @@ def map_dataarray(self, func, x, y, **kwargs):
 
         return self
 
+    def map_dataarray_line(self, x=None, y=None, hue=None, **kwargs):
+        """
+        Apply a line plot to a 2d facet subset of the data.
+
+        Parameters
+        ----------
+        x, y, hue: string
+            dimension names for the axes and hues of each facet
+
+        Returns
+        -------
+        self : FacetGrid object
+
+        """
+        from .plot import line, _infer_line_data
+
+        add_legend = kwargs.pop('add_legend', True)
+        kwargs['add_legend'] = False
+
+        for d, ax in zip(self.name_dicts.flat, self.axes.flat):
+            # None is the sentinel value
+            if d is not None:
+                subset = self.data.loc[d]
+                mappable = line(subset, x=x, y=y, hue=hue,
+                                ax=ax, _labels=False,
+                                **kwargs)
+                self._mappables.append(mappable)
+        _, _, xlabel, ylabel, huelabel = _infer_line_data(
+            darray=self.data.loc[self.name_dicts.flat[0]],
+            x=x, y=y, hue=hue)
+
+        self._finalize_grid(xlabel, ylabel)
+
+        if add_legend and huelabel:
+            self.fig.legend(handles=self._mappables[-1],
+                            labels=list(self.data.coords[huelabel].values),
+                            title=huelabel,
+                            loc="center right")
```

Comment:
It would be great to draw the legend outside the plots, similar to what Seaborn does by default:
![image](https://user-images.githubusercontent.com/1217238/39827831-ff6ad40c-536d-11e8-8132-0a61182bf285.png)

I think the resulting plots look a little cleaner -- otherwise the legend can overlap with the data.

We don't need to copy the full logic from Seaborn (it's pretty complicated!), but its `add_legend()` method could be a useful reference:
https://github.com/mwaskom/seaborn/blob/102cf08f9e40ef7037daf4b3b60b94ee144dc5cb/seaborn/axisgrid.py#L40

This file is already derived from Seaborn, so feel free to copy anything that is relevant.